src/platform_posix.cc

```diff
@@ -368,4 +368,28 @@ optional<std::string> RunExecutable(const std::vector<std::string>& command,
   return result;
 }
 
+optional<std::string> GetGlobalConfigDirectory() {
+  char const* xdg_config_home = std::getenv("XDG_CONFIG_HOME");
+  char const* home = std::getenv("HOME");
+  std::string config;
+
+  // If it exists use XDG_CONFIG_HOME to comply with the XDG base
+  // directory spec. Otherwise, use HOME if available.
+  if (xdg_config_home) {
+    config = xdg_config_home;
+    EnsureEndsInSlash(config);
+    config += "cquery";
+    if (!FileExists(config)) {
+      MakeDirectoryRecursive(AbsolutePath(config, true));
```

Comment:
```
MakeDirectoryRecursive(AbsolutePath(config, false /*validate*/));
```

If validate is true then cquery may abort since validate means the directory should exist.

Reply:
Good catch, have fixed it up